In [1]:
import pandas as pd
import numpy as np

credit_df = pd.read_csv("../data/processed/credit_cleaned.csv")

credit_df.head()

,Unnamed: 0,Age,Sex,Job,Housing,Saving accounts,Checking account,Credit amount,Duration,Purpose,Risk
0,0,67,male,2,own,Unknown,little,1169,6,radio/TV,good
1,1,22,female,2,own,little,moderate,5951,48,radio/TV,bad
2,2,49,male,1,own,little,Unknown,2096,12,education,good
3,3,45,male,2,free,little,little,7882,42,furniture/equipment,good
4,4,53,male,2,free,little,little,4870,24,car,bad


In [2]:
credit_df["Risk"].value_counts()


Risk
good    700
bad     300
Name: count, dtype: int64

In [3]:
credit_df["Risk"].value_counts(normalize=True)

Risk
good    0.7
bad     0.3
Name: proportion, dtype: float64

In [4]:
credit_df["Risk"].value_counts(normalize=True) * 100

Risk
good    70.0
bad     30.0
Name: proportion, dtype: float64

In [5]:
credit_df.columns

Index(['Unnamed: 0', 'Age', 'Sex', 'Job', 'Housing', 'Saving accounts',
       'Checking account', 'Credit amount', 'Duration', 'Purpose', 'Risk'],
      dtype='str')

In [6]:
credit_df.drop(columns="Unnamed: 0", inplace=True)

credit_df.to_csv("../data/processed/credit_cleaned.csv", index=False)

In [14]:
credit_df = pd.read_csv("../data/processed/credit_cleaned.csv")

credit_df.columns

Index(['Age', 'Sex', 'Job', 'Housing', 'Saving accounts', 'Checking account',
       'Credit amount', 'Duration', 'Purpose', 'Risk'],
      dtype='str')

In [15]:
print(f"P(Good Risk) = {P_good:.2f}")
print(f"P(Bad Risk) = {P_bad:.2f}")

P(Good Risk) = 0.70
P(Bad Risk) = 0.30


## Conditional Probability

### Business Question

What is the probability that a customer is classified as bad risk **given that** the loan duration is greater than 24 months?

### Formula

P(Bad Risk | Duration > 24)

### Business Insight

This conditional probability helps the bank understand whether longer loan durations are associated with increased credit risk.

In [16]:
long_duration = credit_df[credit_df["Duration"] > 24]

In [17]:
len(long_duration)

230

In [18]:
bad_long_duration = long_duration[long_duration["Risk"] == "bad"]

In [19]:
len(bad_long_duration)

102

In [20]:
P_bad_given_long = len(bad_long_duration) / len(long_duration)

print(f"P(Bad Risk | Duration > 24 months) = {P_bad_given_long:.3f}")

P(Bad Risk | Duration > 24 months) = 0.443


## Conditional Probability: Loan Duration

### Business Question

What is the probability that a customer is classified as bad risk given that the loan duration is greater than 24 months?

### Calculation

- Customers with duration > 24 months = **230**
- Bad-risk customers among them = **102**

P(Bad Risk | Duration > 24 months) = 102 / 230 = **0.443**

### Business Insight

The overall probability of bad risk is **30%**, but it increases to **44.3%** for customers with loan durations greater than 24 months.

This suggests that longer loan durations are associated with higher credit risk and should be considered during credit evaluation.

In [21]:
high_credit = credit_df[credit_df["Credit amount"] > 5000]

In [22]:
len(high_credit)


188

In [23]:
bad_high_credit = high_credit[high_credit["Risk"] == "bad"]

In [24]:
len(bad_high_credit)

78

In [25]:
P_bad_given_high_credit = len(bad_high_credit) / len(high_credit)

print(f"P(Bad Risk | Credit Amount > 5000) = {P_bad_given_high_credit:.3f}")

P(Bad Risk | Credit Amount > 5000) = 0.415


## Conditional Probability: Credit Amount

### Business Question

What is the probability that a customer is classified as bad risk given that the credit amount is greater than 5000?

### Calculation

- Customers with credit amount > 5000 = **188**
- Bad-risk customers among them = **78**

P(Bad Risk | Credit Amount > 5000) = 78 / 188 = **0.415**

### Business Insight

The probability of bad risk increases from **30%** overall to **41.5%** among customers requesting loans greater than 5000.

This indicates that larger loan amounts are associated with a higher probability of bad credit risk.

In [26]:
young_customers = credit_df[credit_df["Age"] < 30]

In [27]:
len(young_customers)

371

In [28]:
bad_young = young_customers[young_customers["Risk"] == "bad"]

In [29]:
len(bad_young)

137

In [30]:
P_bad_given_young = len(bad_young) / len(young_customers)

print(f"P(Bad Risk | Age < 30) = {P_bad_given_young:.3f}")

P(Bad Risk | Age < 30) = 0.369


## Conditional Probability: Age

### Business Question

What is the probability that a customer is classified as bad risk given that the customer is younger than 30 years?

### Calculation

- Customers aged below 30 = **371**
- Bad-risk customers among them = **137**

P(Bad Risk | Age < 30) = 137 / 371 = **0.369**

### Business Insight

The probability of bad risk increases from **30%** overall to **36.9%** for customers younger than 30 years. This suggests that younger applicants have a moderately higher likelihood of being classified as bad risk in this dataset.

In [31]:
male_customers = credit_df[credit_df["Sex"] == "male"]



In [32]:
len(male_customers)

690

In [33]:
bad_males = male_customers[male_customers["Risk"] == "bad"]

In [34]:
len(bad_males)

191

In [35]:
P_bad_given_male = len(bad_males) / len(male_customers)

print(f"P(Bad Risk | Male) = {P_bad_given_male:.3f}")

P(Bad Risk | Male) = 0.277


## Conditional Probability: Male Customers

### Business Question

What is the probability that a customer is classified as bad risk given that the customer is male?

### Calculation

- Male customers = **690**
- Bad-risk male customers = **191**

P(Bad Risk | Male) = 191 / 690 = **0.277**

### Business Insight

The probability of bad risk among male customers is **27.7%**, which is slightly lower than the overall probability of **30%**.

This suggests that gender alone is not a strong indicator of credit risk in this dataset.

In [36]:
rent_customers = credit_df[credit_df["Housing"] == "rent"]

In [37]:
len(rent_customers)

179

In [38]:
bad_rent = rent_customers[rent_customers["Risk"] == "bad"]

In [39]:
len(bad_rent)

70

In [40]:
P_bad_given_rent = len(bad_rent) / len(rent_customers)

print(f"P(Bad Risk | Housing = Rent) = {P_bad_given_rent:.3f}")

P(Bad Risk | Housing = Rent) = 0.391


## Conditional Probability: Housing = Rent

### Business Question

What is the probability that a customer is classified as bad risk given that the customer lives in rented housing?

### Calculation

- Customers living in rented housing = **179**
- Bad-risk customers among them = **70**

P(Bad Risk | Housing = Rent) = 70 / 179 = **0.391**

### Business Insight

Customers living in rented housing have a **39.1%** probability of being classified as bad risk, which is higher than the overall probability of **30%**.

In [41]:
probability_summary = pd.DataFrame({
    "Condition": [
        "Overall",
        "Duration > 24 months",
        "Credit Amount > 5000",
        "Age < 30",
        "Male",
        "Housing = Rent"
    ],
    "Probability of Bad Risk": [
        0.300,
        0.443,
        0.415,
        0.369,
        0.277,
        0.391
    ]
})

probability_summary

,Condition,Probability of Bad Risk
0,Overall,0.300
1,Duration > 24 months,0.443
2,Credit Amount > 5000,0.415
3,Age < 30,0.369
4,Male,0.277
5,Housing = Rent,0.391


### probability_summary

# Probability Analysis Summary

The conditional probability analysis shows that the probability of bad credit risk increases under certain customer characteristics.

Among the evaluated conditions:

- Duration greater than 24 months has the highest probability of bad risk (44.3%).
- Credit amount greater than 5000 also has a high probability of bad risk (41.5%).
- Customers living in rented housing have a probability of 39.1%.
- Customers younger than 30 years have a probability of 36.9%.
- Male customers have a probability of 27.7%, which is slightly lower than the overall dataset probability.

These findings suggest that loan characteristics such as duration and credit amount are stronger indicators of credit risk than demographic characteristics such as gender.